packages/cozy-client/src/CozyClient.js

```diff
@@ -198,42 +198,51 @@ class CozyClient {
       event: TRIGGER_CREATION
     }
   ) {
-    if (!document._type) return document
-
-    const schema = this.schema.getDoctypeSchema(document._type)
-    if (!schema.cozyMetadata) return document
-
-    const enriched = {
-      ...document
-    }
-    const metadata = Object.entries(schema.cozyMetadata)
-
-    for (const [
-      metadataName,
-      { trigger, value, useCurrentDate = false }
-    ] of metadata) {
-      const shouldSetMetadata =
-        options.event === TRIGGER_CREATION ||
-        (options.event === TRIGGER_UPDATE && trigger === TRIGGER_UPDATE)
-      if (!shouldSetMetadata) continue
-
-      let newValue
-      const metadataPath = `cozyMetadata.${metadataName}`
-
-      const shouldAccumulateValues = Array.isArray(value)
-      if (shouldAccumulateValues) {
-        const prevValue = get(document, metadataPath, [])
-        newValue = uniqBy([...prevValue, ...value])
-      } else if (useCurrentDate) {
-        newValue = new Date().toISOString()
-      } else {
-        newValue = get(document, metadataPath, value)
+    if (this.metadata === undefined) return document
+    let doctypeVersion
+    if (document._type) {
+      const schema = this.schema.getDoctypeSchema(document._type)
+      doctypeVersion = get(schema, 'doctypeVersion')
+    }
+
+    const { slug, sourceAccount = null, version } = this.metadata
```

Comment:
Is there a way to define these values automagicaly in the init if not provided by the calling app ?

Reply:
I don't think so. The client has no idea of what app is initializing it, right?

Reply:
In apps it's available in the DOM (set up by the stack) but this was a true question. I don't know if it's a good thing to let the client fetch that automatically

Reply:
In the case of mobile apps the info is not in the DOM. Mayyybe it could use the current token to look up this information but I think that's a stretch and way outside of the scope of this PR.

Reply:
👌